### PR TITLE
Fix age-restricted videos

### LINKED ISSUE #581: Age restricted videos are broken again
### Version

6.0.4 & master

### Details

With the removal of `get_video_info` we currently don't have a way to resolve age-restricted videos anymore.

### Steps to reproduce

```csharp
// Arrange
var youtube = new YoutubeClient();

// Act
var manifest = await youtube.Videos.Streams.GetManifestAsync("SkRSXFQerZs");

// Assert
manifest.Streams.Should().NotBeEmpty();
```

### PR DESCRIPTION
**Pull request by xBaank**
Closes #581

I also changed one small thing in the tests because tests were failing (your channel now have 78 videos instead of 79).

Another thing was that videos that requires purchase could not throw an exception if purchasePreviewVideoId was not found, so I added a method to check if the reason of not being playable was "This video requires payment to watch.".

### CODE REVIEW

**Comment by Tyrrrz** [comment]
Hi! Thanks a lot for the PR. Can you add `.vs` to `.gitignore` please?

File: YoutubeExplode/Bridge/YoutubeController.cs [hunk 1]
```diff
@@ -124,6 +124,23 @@ internal class YoutubeController
...
+        {
+            string url = "https://www.youtube.com/youtubei/v1/player";
+
+            //Use this as mentioned on https://github.com/Tyrrrz/YoutubeExplode/issues/581#issuecomment-889241520
+            string requestJson = $@"{{""context"": {{""client"": {{""clientName"": ""ANDROID"",""clientScreen"": ""EMBED"",""clientVersion"": ""16.05""}},""thirdParty"": {{""embedUrl"": ""https://www.youtube.com""}}}},""videoId"": ""{videoId}"" }}";
+
+
+            HttpRequestMessage httpRequestMessage = new HttpRequestMessage(HttpMethod.Post,url);
+            httpRequestMessage.Headers.Add("X-Goog-Api-Key", InternalApiKey);
+            httpRequestMessage.Content = new StringContent(requestJson, System.Text.Encoding.UTF8, "application/json");
```
Comment (Tyrrrz) [inline YoutubeExplode/Bridge/YoutubeController.cs:137]: Can we rewrite it to use object serialization, similar to `GetPlaylistAsync(...)` method? The string is a bit hard to read.

https://github.com/Tyrrrz/YoutubeExplode/blob/314be1cade649064ba8f3e6620b812a4ea651539/YoutubeExplode/Bridge/YoutubeController.cs#L160-L187

File: YoutubeExplode/Videos/Streams/StreamClient.cs [hunk 2]
```diff
@@ -194,17 +195,22 @@ public StreamClient(HttpClient httpClient)
...
+
             if (playerResponseFromWatchPage is not null)
             {
+                if (playerResponseFromWatchPage.TryGetAgeRestricted())
+                    //if it's null, it will continue with the previous playerResponse
+                    playerResponseFromWatchPage = await _controller.GetPlayerResponseFromEndpoint(videoId) ?? playerResponseFromWatchPage;
+
                 var purchasePreviewVideoId = playerResponseFromWatchPage.TryGetPreviewVideoId();
-                if (!string.IsNullOrWhiteSpace(purchasePreviewVideoId))
+                bool isPurchaseRequired = playerResponseFromWatchPage.TryGetPurchaseRequired();
```
Comment (Tyrrrz) [inline YoutubeExplode/Videos/Streams/StreamClient.cs:206]: Do we need `isPurchaseRequired` in addition to `purchasePreviewVideoId`? It seems if the latter is not null, it's fair to assume that the video is age-restricted.

File: YoutubeExplode/Bridge/Extractors/VideoWatchPageExtractor.cs [hunk 3]
```diff
@@ -89,6 +88,8 @@ internal partial class VideoWatchPageExtractor
                 .Pipe(j => new PlayerResponseExtractor(j))
         );
 
+        public static PlayerResponseExtractor? TryGetPlayerResponse(string json) => String.IsNullOrWhiteSpace(json) ? null : new PlayerResponseExtractor(Json.Parse(json));
```
Comment (Tyrrrz) [inline YoutubeExplode/Bridge/Extractors/VideoWatchPageExtractor.cs:91]: Can `json` ever be null or whitespace here? It seems to be passed from an HTTP request where it's reasonable to assume it's always going to be non-whitespace.

**Comment by Tyrrrz** [comment]
Also: looks like the build failed on CI

**Comment by xBaank** [comment]
I removed isPurchaseRequired as is not needed (I don't know why i added it, because now it's finding the previewVideoId), changed it to use object serialization, removed IsNullOrWhiteSpace of the json and other things so it passes the build.

File: YoutubeExplode/Bridge/Extractors/VideoWatchPageExtractor.cs [hunk 4]
```diff
@@ -127,6 +128,9 @@ internal partial class VideoWatchPageExtractor
         {
             var content = Html.Parse(raw);
 
+            if (content is null)
+                return null;
+
             var isValid = content.Body.QuerySelector("#player") is not null;
```
Comment (Tyrrrz) [inline YoutubeExplode/Bridge/Extractors/VideoWatchPageExtractor.cs:134]:
```suggestion
            var isValid = content?.Body?.QuerySelector("#player") is not null;
```

File: YoutubeExplode/Bridge/Extractors/VideoWatchPageExtractor.cs [hunk 5]
```diff
@@ -127,7 +128,7 @@ internal partial class VideoWatchPageExtractor
         {
             var content = Html.Parse(raw);
 
-            var isValid = content.Body.QuerySelector("#player") is not null;
+            var isValid = content?.Body?.QuerySelector("#player") is not null;
```
Comment (Tyrrrz) [inline YoutubeExplode/Bridge/Extractors/VideoWatchPageExtractor.cs:131]:
```suggestion
            var isValid = content.Body?.QuerySelector("#player") is not null;
```

**Pull request merged**